### PR TITLE
On Linux store rd-server.json and credential-server.json in apphome.

### LINKED ISSUE #5419: paths_linux.go: `AppHome` should be `filepath.Join(dataHome, appName)`
### Actual Behavior

it's too confusing with `AppHome:  filepath.Join(configHome, appName),`

On macOS, we're correct putting `settings.json` in `~/Library/Preferences/rancher-desktop`

On linux, `credential-server.json` and `rd-engine.json` should go in `~/.local/share/rancher-desktop`

and the factory-reset should delete them from the old location (which will happen if `rdctl factory-reset` deletes all of `~/.config/rancher-desktop`

### PR DESCRIPTION
**Pull request by ericpromislow**
Fixes #5419

And 'apphome' is ~/.local/share/rancher-desktop

### CODE REVIEW

File: pkg/rancher-desktop/config/settingsImpl.ts
Comment (jandubois) [inline pkg/rancher-desktop/config/settingsImpl.ts:407]: Conceptually it seems just wrong to me that a schema migration can have side-effects that modify the filesystem.

A schema migration should just transform a JSON object into another JSON object and not have any observable side-effect.

E.g. the migration code should be called on `settings` API requests and deployment profiles as well (see #4939), and having those potentially touch the filesystem is just wrong.

**Review by jandubois — CHANGES_REQUESTED** [review]
I haven't reviewed the rest of the PR, but I wanted to register my strong objection to deleting files as a side effect of a settings schema migration.

The old files can simply be left behind. They should be deleted by a factory-reset, but otherwise they are not getting in the way, and are of trivial size, so just leave them alone!

File: pkg/rancher-desktop/config/settingsImpl.ts
Comment (ericpromislow) [reply pkg/rancher-desktop/config/settingsImpl.ts]: I finally reverted that code -- thought I did that last week.

File: pkg/rancher-desktop/utils/__tests__/paths.spec.ts
Comment (jandubois) [inline pkg/rancher-desktop/utils/__tests__/paths.spec.ts:137]: Not sure about the point of this test. Isn't is effectively verifying that

```
paths.lima === filepath.Join(paths.appHome, "lima")
```

Not sure why we need to check all the other directories. Not even sure if we need this test at all.

File: src/go/rdctl/pkg/config/config.go
Comment (jandubois) [inline src/go/rdctl/pkg/config/config.go:76]: Not sure why `configDir` is a global variable; it seems be only used inside this function.

But I'm also confused that it is called `configDir` and not `dataDir` or `appDataDir` to make it clearer that it is no longer pointing to `os.UserConfigDir()`.

But that opens up the question if `--config-path` would have to change too, which I don't think is necessary.

File: pkg/rancher-desktop/utils/__tests__/paths.spec.ts
Comment (ericpromislow) [reply pkg/rancher-desktop/utils/__tests__/paths.spec.ts:137]: Yeah, we no longer need this test now that `paths.appHome` means the same thing on linux and macOS.

File: src/go/rdctl/pkg/config/config.go
Comment (ericpromislow) [reply src/go/rdctl/pkg/config/config.go:76]: Question 1:
The docstring for `DefineGlobalFlags`:
```
// DefineGlobalFlags sets up the global flags, available for all sub-commands
```

`configDir` points to the location of `rd-engine.json`. Users can override it with the `--config-path` option.

I think some of the confusion in this whole appHome/DataHome/config area is from the definition of `os.UserConfigDir()` in the Go world. On Linux it points to `${XDG_CONFIG_HOME:-$HOME/.config}` but on macOS it points to `$HOME/Library/Application Support`

File: bats/tests/containers/factory-reset.bats
Comment (adamkpickering) [inline bats/tests/containers/factory-reset.bats:162]: FWIW, we can change `"$PATH_APP_HOME/lima` to `"$LIMA_HOME"`.

Also, unrelated: what about `PATH_EXTENSIONS`?

**Review by adamkpickering — CHANGES_REQUESTED** [review]

File: src/go/rdctl/pkg/factoryreset/delete_data_linux.go
Comment (ericpromislow) [reply src/go/rdctl/pkg/factoryreset/delete_data_linux.go:34]: My mistake -- I thought `paths.Config` didn't have the `rancher-desktop` part. Note that this is for Electron-specific files, which are stored in `~/.config/Rancher Desktop`. Capital letters and a space.

File: bats/tests/containers/factory-reset.bats
Comment (ericpromislow) [reply bats/tests/containers/factory-reset.bats:162]: `PATH_EXTENSIONS` deleting needs to go into a separate issue. I'm already stretching this a bit for PRs, like replacing calls to `os.UserConfigDir()` with our own `paths` module (which is def better).

**Review by jandubois — DISMISSED** [review]
Looks good to me, but I only read the code; I did not run it.

There is one bug that needs to be fixed: `$LIMA_HOME/lima` → `$LIMA_HOME`.

File: bats/tests/containers/factory-reset.bats
Comment (jandubois) [reply bats/tests/containers/factory-reset.bats]: Can you create an issue for dealing with `$PATH_EXTENSIONS`?

Comment (ericpromislow) [reply bats/tests/containers/factory-reset.bats]: Done in #5599

**Review by jandubois — APPROVED** [review]
Thanks, LGTM.

Note that I only reviewed the code, but didn't test the PR on Linux. Merging should be done after @adamkpickering approves it.

**Comment by jandubois** [comment]
Not sure why this PR was moved back to "draft".

Afaict all requested changes from me and @adamkpickering have been applied.

**Comment by ericpromislow** [comment]
I moved it to draft because I thought you wanted the windows APPDATA => LOCALAPPDATA change in first. They're related, but independent of one another.

**Comment by ericpromislow** [comment]
Latest change just removed the import-cycle from windows-specific Go code.

File: src/go/rdctl/pkg/factoryreset/delete_data.go
Comment (adamkpickering) [inline src/go/rdctl/pkg/factoryreset/delete_data.go:103]: We should check `err` here.

**Review by adamkpickering — DISMISSED** [review]
Only one comment really, and I don't think it is imperative to address it.

I did some basic testing on Linux and things seem to work as they should.

File: src/go/rdctl/pkg/config/config.go
Comment (jandubois) [inline src/go/rdctl/pkg/config/config.go:70]: In `cmd/shell.go` you are importing `paths.go` as `p` and use `paths` as the local variable. Here you import it by the normal name and use `appPaths` for the variable instead.

I think this should be handled consistently. I'm not a fan of using `p` for the import, so using `appPaths` seems better to me.

I would even go ahead and change `paths.GetPaths()` to `paths.GetAppPaths()` to match, but maybe that is too much for this PR?

**Review by adamkpickering — DISMISSED** [review]

**Review by adamkpickering — DISMISSED** [review]

**Review by adamkpickering — APPROVED** [review]

**Review by jandubois — APPROVED** [review]
I have various stylistics concerns, but I want the functionality merged to unblock other PRs. I'm approving under the assumptions that the concerns in #5629 are going to be addressed in a follow-up PR.

**Pull request merged**
